### DISCUSSION TITLE
Does ruby-kakfa supports closest replica fetch feature?

### DISCUSSION THREAD

**Discussion by idanto** [zendesk/ruby-kafka#867] [Q&A] [answered]
Kafka today supports the closest replica fetch for consumers which can improve performance and save network traffic in case you have a multi-rack cluster.

https://cwiki.apache.org/confluence/display/KAFKA/KIP-392%3A+Allow+consumers+to+fetch+from+closest+replica

Does this library support this feature? I can't find any evidence for that in the documentation.

**Comment by geoff2k** [comment]
It does not.   KIP-392 requires a Fetch version 11 request that supports passing in the `rack_id` (https://kafka.apache.org/protocol#The_Messages_Fetch), and ruby-kafka uses a version 4 request:

https://github.com/zendesk/ruby-kafka/blob/master/lib/kafka/protocol/fetch_request.rb#L41

**Comment by idanto** [comment]
Thanks for the quick answer @geoff2k.

Do you know if there are any plans to support it? or why ruby-kafka stays with version 4 request - I assume for backward compatibility for older Kafka version but maybe we can add support in newer request versions.

**Comment by dasch** [comment, marked as answer]
Backwards compatibility is the main answer. Ruby-kafka does not have multi-version Kafka support, so a decision to update an API version means that backwards compatibility is lost.
